common/src/main/scala/models/JsonUtils.scala

```diff
@@ -34,4 +38,12 @@ object JsonUtils {
     override def writes(o: Option[T]): JsValue =
       o map format.writes getOrElse JsNull
   }
+
+  implicit val urlFormat = new Format[URL] {
+    override def writes(o: URL): JsValue = JsString(o.toExternalForm)
+    override def reads(json: JsValue): JsResult[URL] = json match {
+      case JsString(url) => Try(JsSuccess(new URL(url))).getOrElse(JsError(s"Invalid url: $url"))
```

Comment:
Do we really need java.net.URL semantics here ? It establishes dangerous stream handler that will resolve actual Internet host  on .equals for instance.